data-prepper-api/src/main/java/org/opensearch/dataprepper/typeconverter/StringConverter.java

```diff
@@ -25,6 +27,9 @@ public String convert(Object source) throws IllegalArgumentException {
         if (source instanceof String) {
             return (String)source;
         }
+        if (source instanceof BigDecimal) {
+            return ((BigDecimal)source).toPlainString();
```

Comment:
If I understand correctly, `toString()` would result in a form same or closer to the source but `toPlainString` would do a conversion to remove scientific notation. In that case, I would prefer `toString()` to avoid surprising the user with the conversion. For example, `1.23E+308` results in a string with more than 300 characters.


And then we can combine everything under:
```java
if (source instanceof Number || source instanceof Boolean || source instanceof String)  {
    return source.toString();
}
```

Reply:
Initially, I thought that `toPlainString()` would help in getting the exact value, in case someone wants to have it. But agree on the above, made the change for it.